packages/components/src/components/buttons/ButtonGroup/ButtonGroup.tsx

```diff
@@ -63,19 +76,41 @@ export const ButtonGroup = ({
 
     if (!areChildrenValid) {
         console.error(
-            'Invalid children passed to ButtonGroup. Only Button and IconButton are allowed.',
+            'Invalid children passed to ButtonGroup. Only Button, IconButton, and Tooltip (containing Button or IconButton) are allowed.',
         );
 
         return null;
     }
 
-    const childrenWithProps = children.map(child =>
-        React.cloneElement(child, {
-            variant: child.props.variant || variant,
-            size: child.props.size || size,
-            isDisabled: child.props.isDisabled || isDisabled,
-        }),
-    );
+    const childrenWithProps = React.Children.map(children, child => {
+        if (React.isValidElement(child)) {
+            if (
+                child.type === Tooltip &&
+                React.isValidElement((child.props as TooltipProps).children)
+            ) {
+                const tooltipProps = child.props as TooltipProps;
+                const tooltipChild = tooltipProps.children as React.ReactElement;
+                const tooltipChildProps: ButtonProps | IconButtonProps = tooltipChild?.props;
```

Comment:
lets extract this to ;AllowedChildrenPropsType` so it easier to add another one and it at one place